astropy/utils/compat/optional_deps.py

```diff
@@ -11,8 +11,9 @@
 _optional_deps = ['bleach', 'bottleneck', 'bs4', 'bz2', 'h5py', 'html5lib',
                   'IPython', 'jplephem', 'lxml', 'matplotlib', 'mpmath',
                   'pandas', 'PIL', 'pytz', 'scipy', 'skyfield',
-                  'sortedcontainers', 'lzma', 'yaml']
```

Comment:
This removes `HAS_YAML` from the namespace without any deprecation... Is this okay?

Reply:
I keep it around with the `_formerly_optional_deps`

Reply:
But there is no deprecation. Do we plan to just keep it around forever?

Reply:
Should we even deprecate `HAS_YAML`?

Reply:
🤷

Reply:
Hmm, @pllim is probably right that we should deprecate, and 5.0 is a good release to do it.

Reply:
I'll add a deprecation warning.